qtop.py

```diff
@@ -1680,9 +1691,12 @@ def safe_exit_with_file_close(handle, name, stdout, delete_file=False):
                 os.chdir(QTOPPATH)
                 unlink(name)
 
+            if options.TAR:
+                add_to_tar(name, config['savepath'])
         except (KeyboardInterrupt, EOFError):
             safe_exit_with_file_close(handle, name, stdout)
         else:
-            pass
+            if options.TAR >= 1:
+                add_to_tar(QTOP_LOGFILE, config['savepath'])
```

Comment:
all good, just generalise the functions to make them amenable to zip instead of tar etc (somebody else's job, but still, let's have it generic)